### ISSUE TITLE
Untrained words influence Tensorflow Embedding 

### ISSUE THREAD

**Issue by ctrado18** [RasaHQ/rasa_nlu#1319]
Hi,

I see some strange things that unseen words within intent classification are changing the intent. Although  Two  diffeernt words (not in training data for intent) give two different intents!
But, one of the word is used for NER_CRF training where I set the intent label to`""` and I thought this is then not used for intent training.  Therefore, it seems that the tensorflow embedding knows this words although the intent label was removed?!

I have a very strong suggestion. Is it possible that the order of the NER_CRF pipeline and the tensorflow embedding pipeline plays a role?

Here the config file:

```

language: "de"


pipeline:
- name: "tokenizer_whitespace"
- name: "ner_crf"
- name: "intent_featurizer_count_vectors"
  "stop_words": ['mein','der','die','das','ein']
- name: "intent_classifier_tensorflow_embedding"
  intent_tokenization_flag: false
  intent_split_symbol: "+"
```

chagning NER_CRF to the end,  yields totally different and correct result such that the word used for entity training is not seen by tensorflow embedding!?

```

language: "de_core_news_sm"


pipeline:
- name: "intent_featurizer_count_vectors"
  "stop_words": ['mein','der','die','das','ein']
- name: "intent_classifier_tensorflow_embedding"
  intent_tokenization_flag: false
  intent_split_symbol: "+"
- name: "tokenizer_whitespace"
- name: "ner_crf"

```

**Apart from above: I tried now same sentence with few OOVs and everytime I get different intents! SO there seems really an issue.**

Maybe some of default setting in a pipeline like intent_featurizer_count_vectors does the thing. Maybe the standard `token_pattern` plays a role? What is the meaning of it?

**Comment by Ghostvv** [comment]
Could you please compile a small example, so I could reproduce it?

**Comment by Ghostvv** [comment]
I tried it with our datasets, and couldn't reproduce it. I get the expected behavior with and without `ner_crf`

**Comment by ctrado18** [comment]
> Could you please compile a small example, so I could reproduce it?

I will try.

But this should be independent the order of both pielines and there are no interdependences? 

I tried it now few times after each other and get everytime different results. So maybe my system is trapped in an unstable point like a saddlepoint. Maybe I have too less data and for this sentence all intents share 1 or 2 examples of this sentence. Therefore the intent switches randomly? 
Could I use the OOV attribute since I have in this case `sent + OOV`. But I would like to have a general OOV not some special examples in the OOV list. Is this possible without a OOV list?

**Comment by Ghostvv** [comment]
Since I cannot reproduce it, It's hard to tell what's going on in your case.
What do you mean by general OOV? During prediction time all unseen words are treated as OOV, but during training the algorithm needs this OOV token to be present in training data either explicitly or as list of words that should be treated as OOV.

**Comment by ctrado18** [comment]
Yeah would like to add this sentence to my data but together with OOV.

So, I can put to my data this sentence and append just the string `OOV`? So the algo knows what to do in this case.

**Comment by Ghostvv** [comment]
yes, you can just use `OOV` instead of a word. But you'd need to specify `OOV_token: "OOV"` in your config file

**Comment by ctrado18** [comment]
Ok. If I have no word lost for OOV,I set

```
 # will be converted to lowercase if lowercase is true
  "OOV_token": OOV# string or None
  "OOV_words": []  # list of strings
``

Now examples in training data can be like `hello stupid thing OOV`. Would that the correct usage?

**Comment by Ghostvv** [comment]
both

**Comment by Ghostvv** [comment]
I'll close the issue since original problem seems to be resolved
